soft4pes/model/grid/rl_grid.py

```diff
@@ -0,0 +1,163 @@
+        grid current in alpha-beta frame.
+    """
+
+    def __init__(self, Vgr, fgr, Rg, Lg, base):
+        """
+        Initialize a RLGrid instance. Set inital state to zero.
+
+        Parameters
+        ----------
+        Vgr : float
+            Grid rated voltage [V].
+        fgr : float
+            Grid rated frequency [Hz].
+        Rg : float
+            Resistance [Ohm].
+        Lg : float
+            Inductance [H].
+        base : base value object
+            Base values.
+        """
+        self.Vgr = Vgr / base.V
+        self.wg = 2 * np.pi * fgr / base.w
+        self.Rg = Rg / base.Z
+        self.Xg = Lg / base.L
+        self.x = np.array([0, 0])
+        self.base = base
+
+    def get_next_state(self, x, u, v_dc, t, Ts):
+        """
+        Calculate the next state of the grid using discrete state-space model.
+
+        Parameters
+        ----------
+        x : 1 x 2 ndarray of floats
+            Current state of the grid [pu].
+        u : 1 x 3 ndarray of ints
+            Converter three-phase switch position.
+        v_dc : float
+            Converter dc-link voltage [pu].
+        t : float
+            Current time [s].
+        Ts : float
+            Time step [s].
+
+        Returns
+        -------
+        1 x 2 ndarray of floats
+            Next state of the grid [pu].
+        """
+
+        matrices = self.get_state_space(v_dc, Ts)
+
+        vg = self.get_grid_voltage(t)
+
+        return np.dot(matrices['A'], x) + np.dot(matrices['B1'], u) + np.dot(
+            matrices['B2'], vg)
+
+    def get_state_space(self, v_dc, Ts):
+        """
+        Get the discrete state-space model of the grid in alpha-beta frame.
+        Discretization is done using the forward Euler method.
+
+        Parameters
+        ----------
+        v_dc : float
+            Converter dc-link voltage [pu].
+        Ts : float
+            Time step [s].
+
+        Returns
+        -------
+        dict
+            A dictionary containing matrices A, B1, B2, and C of the state-space model.
+        """
+        Rg = self.Rg
+        Xg = self.Xg
+        Ts = Ts * self.base.w
+
+        # Clarke transformation matrix
+        K = (2 / 3) * np.array([[1, -1 / 2, -1 / 2],
+                                [0, np.sqrt(3) / 2, -np.sqrt(3) / 2]])
+
+        F = -Rg / Xg * np.eye(2)
+        G1 = v_dc / 2 * 1 / Xg * K
+        G2 = -1 / Xg * np.eye(2)
+
+        A = np.eye(2) + F * Ts
+        B1 = G1 * Ts
+        B2 = G2 * Ts
+        C = np.array([[1, 0], [0, 1]])
+
+        return {'A': A, 'B1': B1, 'B2': B2, 'C': C}
+
+    def get_grid_voltage(self, t):
+        """
+        Get the grid voltage at a specific time instant.
+
+        Parameters
+        ----------
+        t : float
+            Current time [s].
+
+        Returns
+        -------
+        1 x 2 ndarray of floats
+            Grid voltage in alpha-beta frame [pu].
+        """
+
+        theta = self.wg * (t * self.base.w)
```

Comment:
this projection is not very convenient and straightforward. We can find the future fundamental quantities (e.g., grid voltage) with the following steps:

- transform the current abc quantity into the alpha-beta frame
- take the alpha-beta vector and rotate it (counter-clockwise, or clockwise, depending on the variable of interest) by w * Ts

Alternatively, this can be done with the variable in dq which is transformed into alpha-beta

Reply:
So you would prefer to have the voltages as a state-like variable, that is saved and updated?

Reply:
This does not have to do with whether a variable is part of the state or a disturbance. It is about how it is handled when trying to find its future value. It is best to do the rotation it in the alpha-beta frame

Reply:
Yes, that I understand. I will rephrase: Can the current time still be used to get the voltages in abc-frame every time the function is called, or do you want the voltages to be initialized with the class instance, and then rotated and updated? (I prefer the first option)